R/Solver.R

```diff
@@ -9,161 +9,197 @@
 ### or it can be a matrix of column vectors to solve for.
 ### If Rank_Analysis has been done, the rank_an output can be inputted to avoid repeating rank-finding.
 ### DFT_obj is an output from the gen_DFT() function
-solve_WA <- function(DFT_obj,A_mat,v,
-                     rank_obj=NULL,DID_full=FALSE) {
+solve_WA <- function(
+    DFT_obj,
+    A_mat,
+    v,
+    rank_obj = NULL,
+    DID_full = TRUE
+) {
   ## If v is a vector, put it into matrix form:
   if (is.vector(v)) {
-    v <- matrix(data=v, ncol=1)
+    v <- matrix(data = v, ncol = 1)
   }
-  
+
   F_mat <- DFT_obj$F_mat
-  
+
   if (!is.null(rank_obj)) { # if given a rank object, use it:
     if (ncol(v) != ncol(rank_obj$FTv_Ranks)) {
-      stop(simpleError("v and rank_obj do not come from the same dimension of v."))
+      stop("v and rank_obj do not come from the same dimension of v.")
     }
+
     FT_qr <- rank_obj$FT_qr
     FT_rank <- FT_qr$rank
     RankAT <- rank_obj$RankAT
-    if (sum(rank_obj$FTv_Ranks["Dim_W",] < 0)==ncol(v)) {
-      stop(simpleError("No columns of v have solutions."))
-    } else if (sum(rank_obj$FTv_Ranks["Dim_W",] < 0) > 0) {
-      DefCols <- rank_obj$FTv_Ranks["Dim_W",] < 0
-      warning(simpleWarning(paste0("The following columns of v have no solutions and were dropped: ",
-                                   paste((1:ncol(rank_obj$FTv_Ranks))[DefCols], collapse=", "),
-                                   ". The remaining columns have been shifted accordingly. Interpret results accordingly.")))
-      v <- v[,!DefCols,drop=FALSE]
+
+    # Check for solvability
+    no_solution <- rank_obj$FTv_Ranks["Dim_W", ] < 0
```

Comment:
`rank_obj$FTv_Ranks["Dim_W",] < 0` is repeated, so I'm storing it in a variable, `no_solution`.